src/Shared/LayoutRenderers/AspNetAppBasePathLayoutRenderer.cs

```diff
@@ -23,80 +23,87 @@ namespace NLog.Web.LayoutRenderers
     /// </summary>
 #else
     /// <summary>
-    /// Rendering Application BasePath. <see cref="HostingEnvironment.MapPath"/>("~")
+    /// Rendering Application BasePath. <see cref="IHostEnvironment.MapPath"/>("~")
     /// </summary>
 #endif
     [LayoutRenderer("aspnet-appbasepath")]
     [ThreadAgnostic]
     public class AspNetAppBasePathLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
-        private IHostEnvironment HostEnvironment => _hostEnvironment ?? (_hostEnvironment = ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration));
+        /// <summary>
+        /// Context for DI
+        /// </summary>
         private IHostEnvironment _hostEnvironment;
 
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostEnvironment?.ContentRootPath, out _appBasePath);
-        private string _appBasePath;
-#else
-        private string AppBasePath => _appBasePath ?? ResolveAppBasePath(HostingEnvironment.MapPath("~"), out _appBasePath);
-        private static string _appBasePath;
-#endif
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        public IHostEnvironment HostEnvironment
+        {
+            get => _hostEnvironment ?? (_hostEnvironment = RetrieveHostEnvironment(ResolveService<IServiceProvider>(), LoggingConfiguration));
+            set => _hostEnvironment = value;
+        }
 
-        /// <inheritdoc />
-        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        internal static IHostEnvironment RetrieveHostEnvironment(IServiceProvider serviceProvider, LoggingConfiguration loggingConfiguration)
         {
-            builder.Append(AppBasePath);
+            return ServiceLocator.ResolveService<IHostEnvironment>(serviceProvider, loggingConfiguration);
         }
+#else
+        /// <summary>
+        /// Provides access to the current IHostEnvironment
+        /// </summary>
+        /// <returns>IHostEnvironment or <c>null</c></returns>
+        [NLogConfigurationIgnoreProperty]
+        public IHostEnvironment HostEnvironment { get; set; } = new HostEnvironment();
```

Comment:
Can this become `internal` and changed so `HostEnvironment`-class has a static-property that points to a `Default`-instance:

`public IHostEnvironment HostEnvironment { get; set; } = HostEnvironment.Default;`

Reply:
Complete. forgot I did the request body middleware options the same way